hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java

```diff
@@ -196,8 +196,8 @@ private List<String> getPartitionPathWithPathPrefixUsingFilterExpression(String
                     } else if (!path.getName().equals(HoodieTableMetaClient.METAFOLDER_NAME)) {
                       return Pair.of(Option.empty(), Option.of(path));
                     }
-                  } else if (path.getName()
-                      .startsWith(HoodiePartitionMetadata.HOODIE_PARTITION_METAFILE_PREFIX)) {
+                  } else if (path.getName().startsWith(HoodiePartitionMetadata.HOODIE_PARTITION_METAFILE_PREFIX)
+                      || MetadataPartitionType.isValidPartitionFilePath(path.getName())) {
```

Comment:
Let's not add MDT specific logic in this method, which should not be aware of whether the table is a data table or metadata table.  Could we fix the the issue of writing partition metadata file in MDT directly?